packages/core/src/external/carequality/ihe-gateway-v2/report/report.ts

```diff
@@ -0,0 +1,321 @@
+import { Pool } from "pg";
+import { z } from "zod";
+import dayjs from "dayjs";
+import {
+  OutboundPatientDiscoveryResp,
+  XCPDGateway,
+  OutboundDocumentQueryResp,
+  XCAGateway,
+  OutboundDocumentRetrievalResp,
+  outboundPatientDiscoveryRespSchema,
+  outboundDocumentQueryRespSchema,
+  outboundDocumentRetrievalRespSchema,
+} from "@metriport/ihe-gateway-sdk";
+
+const patientDiscoveryReportRequests = 10;
+const documentQueryReportRequests = 100;
+const documentRetrievalReportRequests = 100;
+
+type Report = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+  errorOids: {
+    oid: string;
+    error: string;
+  }[];
+};
+
+type SimplifiedReport = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+};
+
+type ReportSummary = {
+  simplifiedReports: SimplifiedReport[];
+  averageSuccessPercentage: number;
+  averageFailurePercentage: number;
+  errorDetailsMap: { [key: string]: { count: number; details: string } };
+};
+
+export const patientDiscoveryResultRowSchema = z.object({
+  data: outboundPatientDiscoveryRespSchema,
+});
+
+export const documentQueryResultRowSchema = z.object({
+  data: outboundDocumentQueryRespSchema,
+});
+
+export const documentRetrievalResultRowSchema = z.object({
+  data: outboundDocumentRetrievalRespSchema,
+});
+
+enum Tables {
+  PatientDiscoveryResult = "patient_discovery_result",
+  DocumentQueryResult = "document_query_result",
+  DocumentRetrievalResult = "document_retrieval_result",
+}
+
+enum TableColumns {
+  request_id = "request_id",
+  created_at = "created_at",
+}
+
+async function getRandomRequestIds({
+  pool,
+  tableName,
+  since,
+  limit,
+}: {
+  pool: Pool;
+  tableName: Tables;
+  since: string;
+  limit: number;
+}): Promise<string[]> {
+  const query = `
+    SELECT request_id
+    FROM ${tableName}
+    WHERE created_at >= $1
+    ORDER BY created_at DESC
+    LIMIT $2;
+  `;
+  const values = [since, limit];
+  const { rows } = await pool.query(query, values);
+  return rows.map(row => row.request_id);
```

Comment:
ditto: `row` is probably `any`, right? We could use this instead, to at least make sure we're accessing the correct property:

```ts
... row[TableColumns.requestId]);
```